src/common/snippets/src/pass/split_dimension_m.cpp

```diff
@@ -131,16 +141,23 @@ bool SplitDimensionM::split(const ov::Shape& shape, size_t optimal_parallelism_w
     if (is_prime_number(m_dim))
         return false;
 
-    auto is_optimized = [&](size_t batch_dim) {
-        return batch_dim >= optimal_parallelism_work_amount;
-    };
-
     // We skip optimization if the current batch is optimal for concurrency
-    if (is_optimized(batch_dim))
+    if (batch_dim % optimal_parallelism_work_amount == 0)
         return false;
 
-    std::tie(batch_m_dim, new_m_dim) = get_splited_dimensions(batch_dim, m_dim, optimal_parallelism_work_amount);
-    return is_optimized(batch_dim * batch_m_dim);
+    std::tie(batch_m_dim, new_m_dim) = compute_ideal_cases_heuristic(batch_dim, m_dim, optimal_parallelism_work_amount);
+    if (batch_m_dim != 1)
+        return true;
+
+    // If M dim is big enough, aggressive heuristic is used for kernel_m minimization.
+    // For smaller M dim, conservative heuristic is used to preserve old behavour.
+    const bool big_m_dim = m_dim >= 4000;
```

Comment:
By the way, we also can support the case with small `M`.
If `batch < optimal_parallelism_work_amount` and `M` is quite small (for example, `M` < 64), nothing is needed to be updated and splitted - let's execute as it is.

I don't insist to do it in this PR but I have some models (for example, action-recognition or levit) with small values of batch and `M` where this pass is applied and there will be `M = 4` or even `M = 1`. And this action leads to perf degrdation.

Reply:
I like your idea. In third-party brgemm heuristics, I saw that minimal allowed `m_kernel` is 16. Probably we can take this into account in our heuristics.
But it's also important is that `SplitDimensionM::split` is used in CPU callback (via `can_be_optimized`), so if it returns false, the MHA tokenization doesn't happen. So another question that we need to answer is whether we need to even tokenize such MHA's or not